### ISSUE TITLE
Privacy issue with profile pictures

### ISSUE THREAD

**Issue by VertyyBird** [actuallyaridan/chirp#15]
I know that the current profile pic setup is a temporary solution. But it does open up a privacy issue. People are able to put a shortened URL/ip grabber link into the profile pic and banner fields and they get rendered as images.

For example, I did this with my own profile: https://beta.chirpsocial.net/user/?id=Verty

In the profile picture URL I put `https://l.verty.gay/0ATw`, which gets 301 redirected to `https://verty.gay/wp-content/uploads/2024/10/tanlin.png`
Same for my banner, URL `https://l.verty.gay/H6ed` 301s to `https://verty.gay/wp-content/uploads/2023/11/feature-image.jpg`

Because of this, I can see detailed logs of whoever sees my posts or visits my profile. And since I used a different short URL for my banner, I can tell the difference between a post view and a profile view. This poses a general privacy risk because of this kind of logging, but I'm sure there is also some sort of attack that could be done using this.

There should be some check that the URL provided is a direct link to an image, without any redirection. Or perhaps just whitelist a few domains that images can be hosted on (twitter, imgur, imgbb, etc).

**Comment by ghost** [comment]
I totally agree with you, one way we could fix this would be by proxying all pictures through chirpsocial.net though this could become expensive as the site grows

**Comment by thebrewergame** [comment]
I believe the better approach of what has been suggested here so far is to block redirects and force a direct link instead of whitelisting only a handful of permitted domains. This way, users still have control and more ease of use but can't track specific metrics.

For example, my Chirp profile banners are all stored together in the files for my website in such a way that makes it extremely easy to update them should I choose. I just need to update the image files (but retain the naming structure), and Chirp will update everything accordingly. This is great ease of use for me (and anyone else that wants to replicate it) that could not be done should my domain not be permitted.
